xdsl/ir.py

```diff
@@ -544,6 +534,27 @@ def irdl_definition(cls) -> OpDef:
         ...
 
 
+@dataclass
+class BinaryOperation(Operation):
+    """A generic operation. Operation definitions inherit this class."""
+
+    # TODO replace with trait
+    def verify_(self) -> None:
+        try:
```

Comment:
We can maybe add these asserts in the `__post_init__`, such that this will only be checked once.

Reply:
These can be totally dropped, since we know that we are under a BinaryOperation